w3af/plugins/grep/analyze_cookies.py

```diff
@@ -187,6 +199,27 @@ def _analyze_cookie_security(self, request, response, cookie_obj,
         self._http_only(request, response, cookie_obj,
                         cookie_header_value, fingerprinted)
 
+    def _update_cookie_vulnerability(self, response, vulnerability):
+        # thread lock to escape race condition on db access
+        with self._plugin_lock:
+            vulns = kb.kb.get(self, 'security')
+        
+            if len(vulns) == 0:
+                kb.kb.append(self, 'security', vulnerability)
+            elif len(vulns) >= 1:
+                old_vuln = vulns[0]
```

Comment:
Hmmm... there might be an issue here. Have you tested what happens when a site has cookies which lack the secure AND httponly flags? What I believe will happen is:
 * Plugin receives a response with a cookie without httponly
 * First call to `_update_cookie_vulnerability` will return zero vulns in `vulns = kb.kb.get(self, 'security')`
 * The httponly vulnerability is stored in the kb
 * Call to `_update_cookie_vulnerability` will return the httponly vulnerability we stored in the previous step
 * The new URL (which is vulnerable to secure flag missing) is added to the first vulnerability, which is actually related to httponly